### PR TITLE
Unit test fixes

### PR DESCRIPTION
**Pull request by jsafrane**
I added couple of new unit tests and reworked the tests a bit. See each commit, most of them should be self-explanatory.

"Fix typo in volume name" is special - it fixes a typo, but the fix breaks tests. I am not sure what was the idea behind the broken tests, I made some adjustments so the tests pass and they test something useful.

@gnufied, PTAL

### CODE REVIEW

File: pkg/scheduler/algorithm/predicates/csi_volume_predicate_test.go [hunk 1]
```diff
@@ -226,6 +198,19 @@ func TestCSIVolumeCountPredicate(t *testing.T) {
 			},
 		},
 	}
+	inTreeNonMigratablecsiEBSOneVolPod := &v1.Pod{
```
Comment (gnufied) [inline pkg/scheduler/algorithm/predicates/csi_volume_predicate_test.go]: why do we call this EBS pod? Should we rename this variable to `inTreeNonMigrtableHostPathOneVolPod` or something similar?

Comment (jsafrane) [reply pkg/scheduler/algorithm/predicates/csi_volume_predicate_test.go]: that's actually search-and-replace error, I renamed OneVolPod to csiEBSOneVolPod in the next commit. Fixed.

**Comment by gnufied** [comment]
Changes look reasonable. Please squash them.

Yeah typo was interesting because it did count the volumes correctly but "used" pvc(with typo) did not contribute towards the total count and hence tests worked. Thanks for fixing it.

**Comment by jsafrane** [comment]
Should it be squashed into your unit-test commit?

**Comment by gnufied** [comment]
perhaps yeah.

**Pull request merged**